dsl/api.go

```diff
@@ -108,97 +121,30 @@ func Version(ver string) {
 //    })
 //
 func Docs(fn func()) {
-	docs := new(design.DocsExpr)
+	docs := new(expr.DocsExpr)
 	if !eval.Execute(fn, docs) {
 		return
 	}
 	switch e := eval.Current().(type) {
-	case *design.APIExpr:
+	case *expr.APIExpr:
 		e.Docs = docs
-	case *design.ServiceExpr:
+	case *expr.ServiceExpr:
 		e.Docs = docs
-	case *design.MethodExpr:
+	case *expr.MethodExpr:
 		e.Docs = docs
-	case *design.AttributeExpr:
+	case *expr.AttributeExpr:
+		e.Docs = docs
+	case *expr.HTTPFileServerExpr:
 		e.Docs = docs
 	default:
 		eval.IncompatibleDSL()
 	}
 }
 
-// Contact sets the API contact information. It is used by the generated OpenAPI
-// specification.
-//
-// Contact must appear in a API expression.
-//
-// Contact takes a single argument which is the defining DSL.
-//
-// Example:
-//
-//    var _ = API("divider", func() {
-//        Contact(func() {
-//            Name("support")
-//            Email("support@goa.design")
-//            URL("https://goa.design")
-//        })
-//    })
-//
-func Contact(fn func()) {
-	contact := new(design.ContactExpr)
-	if !eval.Execute(fn, contact) {
-		return
-	}
-	if a, ok := eval.Current().(*design.APIExpr); ok {
-		a.Contact = contact
-		return
-	}
-	eval.IncompatibleDSL()
-}
-
-// License sets the API license. It is used by the generated OpenAPI
-// specification.
-//
-// License must appear in a API expression.
-//
-// License takes a single argument which is the defining DSL.
-//
-// Example:
-//
-//    var _ = API("divider", func() {
-//        License(func() {
-//            Name("MIT")
-//            URL("https://github.com/goadesign/goa/blob/master/LICENSE")
-//        })
-//    })
-//
-func License(fn func()) {
-	license := new(design.LicenseExpr)
-	if !eval.Execute(fn, license) {
-		return
-	}
-	if a, ok := eval.Current().(*design.APIExpr); ok {
-		a.License = license
-		return
-	}
-	eval.IncompatibleDSL()
-}
-
-// TermsOfService sets the terms of service of the API. It is used by the
-// generated OpenAPI specification.
-//
-// TermsOfService must appear in a API expression.
-//
-// TermsOfService takes a single argument which is the TOS text or URL.
-//
-// Example:
-//
-//    var _ = API("github", func() {
-//        TermsOfService("https://help.github.com/articles/github-terms-of-API/"
-//    })
-//
+// TermsOfService describes the API terms of services or links to them.
```

Comment:
Should we keep the header documentation?

Reply:
Fixed